lib/iris/cube.py

```diff
@@ -1600,7 +1600,7 @@ def cell_methods(self, cell_methods):
     def core_data(self):
         """
         Retrieve the data array of this :class:`~iris.cube.Cube` in its
-        current state, which will be either real or lazy.
+        current state, which may be either real or lazy.
```

Comment:
You're going to have to try quite hard to convince me that this is a necessary change. The core data of a cube will _either_ be real or lazy, which my original implies. Changing 'will' to 'may' reads more like 'it might be one of these, but it might be something else instead' 🍌 

We're in the realm of technical documentation here, where absolute clarity is key. Using 'will' gives us that clarity; I'd argue using 'may' does not.